### DISCUSSION TITLE
How to get Immich to wait for external library to mount?

### DISCUSSION THREAD

**Discussion by LearnedLately** [immich-app/immich#13038] [General]
I have an external library, `/mnt/docs/Photos`, that is not loading on startup. The parent folder `/mnt/docs` is mounted on machine startup by a simple systemd service I created, but apparently the docker service is starting first, so the external library is not ready when Immich starts. Weirdly, when Immich can't find this folder, it creates it, which then causes the mount to fail because /mnt/docs isn't empty. I tried adding `Before=docker.service` to my mounting service, but it doesn't seem to work. Anyone have a solution?

**Comment by bo0tzz** [comment]
This isn't really something for Immich to even know about. The systemd ordering seems like the right angle, though I don't know why it wouldn't work.

**Reply by AngelaDMerkel** [reply]
This is pretty easy to do and can be accomplished by delaying the docker engine until later, for example:

```
[Service]
ExecStartPre=/bin/sleep 30
```
